conda_forge_tick/update_upstream_versions.py

```diff
@@ -142,16 +131,57 @@ def get_version(self, url):
         return latest
 
 
-class CRAN(LibrariesIO):
+class CRAN:
+    """The CRAN versions source.
+
+    Uses a local CRAN index instead of one request per package.
+    """
     name = "cran"
     url_contains = "cran.r-project.org/src/contrib/Archive"
+    cran_url = "https://cran.r-project.org"
 
-    def package_name(self, url):
-        return url.split("/")[6]
+    def __init__(self):
+        try:
+            session = requests.Session()
+            self.cran_index = self._get_cran_index(session)
+            logger.info("Cran source initialized")
+        except Exception:
+            logger.error("Cran initialization failed", exc_info=True)
+            self.cran_index = {}
+
+    def _get_cran_index(self, session):
+        # from conda_build/skeletons/cran.py:get_cran_index
```

Comment:
Would it be possible to import this? Should it go somewhere central?

Reply:
The function was renamed and extended in conda-build between current 3.18.9 and master. After the next conda-build release the complete `_get_cran_index` function can be replaced without any additionally refactoring by `from conda_build.skeletons.cran import get_cran_index` :)

Reply:
I am happy to do that in the future, but would prefer keep the code like it is for now.